x/programs/examples/imports/pstate/pstate.go

```diff
@@ -44,12 +44,34 @@ func (i *Import) Name() string {
 func (i *Import) Register(link *host.Link) error {
 	i.meter = link.Meter()
 	wrap := wrap.New(link)
-	if err := wrap.RegisterThreeParamInt64Fn(Name, "put", i.putFn); err != nil {
+	if err := wrap.RegisterAnyParamFn(Name, "put", 3, i.putFnVariadic); err != nil {
 		return err
 	}
-	return wrap.RegisterTwoParamInt64Fn(Name, "get", i.getFn)
+	return wrap.RegisterAnyParamFn(Name, "get", 2, i.getFnVariadic)
 }
 
+func (i *Import) putFnVariadic(caller *program.Caller, args ...int64) (*types.Val, error) {
```

Comment:
we could change our host functions to be variadic functions and throw an error if an incorrect # of params are passed in to avoid this. Would make the host functions less readable.